config/reflection_magic.go

```diff
@@ -97,11 +98,32 @@ func makeArgumentConstructors(fnType reflect.Type, argTypes map[reflect.Type]con
 	return out, nil
 }
 
+func checkConstructorOpts(t reflect.Type, opts ...interface{}) error {
+	if !t.IsVariadic() {
+		if len(opts) > 0 {
+			return errors.New("constructor doesn't accept any options")
+		}
+		return nil
+	}
+	// variadic parameters always go last
+	optType := t.In(t.NumIn() - 1).Elem()
```

Comment:
you use `opt` below, and `optType` had me a bit confused as to which was which. Perhaps call this `wantType` or `variadicType`?

Reply:
Done.